modules/user_accounts/php/edit_user.class.inc

```diff
@@ -304,14 +305,24 @@ class Edit_User extends \NDB_Form
                      'userID'      => $uid,
                     )
                 );
-                $examinerID = $examinerID = $DB->pselectOne(
+                $examinerID = $DB->pselectOne(
                     "SELECT examinerID
-                 FROM examiners
-                 WHERE full_name=:fullName",
+                            FROM examiners
+                            WHERE full_name=:fullName",
                     array('fullName' => $values['Real_name'])
                 );
-            } else {
-
+            } elseif (!empty($examinerID)
+                && ((!empty($ex_radiologist)
+                && !empty($ex_pending)
+                && !empty($ex_curr_sites))
+                || (empty($ex_radiologist)
+                && empty($ex_pending)
+                && empty($ex_curr_sites)))
+            ) {
```

Comment:
fault in logic above, if examiner ID exists, there is no way the examiner pending and approval can be reset to null (the columns in SQL are not nullable, examiners can not be deleted and there is no code below to handle that case) thus the last segment of this statemnt should be removed.

```
|| (empty($ex_radiologist)
                && empty($ex_pending)
                && empty($ex_curr_sites))
```

The possible cases are:
- `!empty($examinerID) && !empty($ex_pending)  && !empty($ex_radiologist)) &&`
    -  `empty($ex_curr_sites)` -> examiner existed but we just removed all its sites
    - `||`
    - `!empty($ex_curr_sites)` -> examiner existed and we changed its sites
- `empty($examinerID) && !empty($ex_pending)  && !empty($ex_radiologist)) &&`
    - `!empty($ex_curr_sites)` -> examiner is new and we added its sites

Reply:
@ridz1208 I changed the logic as you suggested. FYI, the following case is impossible: `!empty($examinerID) && !empty($ex_pending) && !empty($ex_radiologist) && empty($ex_curr_sites)` due to data entry checks. If there are no sites for the examiner, then Radiologist and Pending fields must be emptied out in order for the form to save. That is why I originally added the second "or" with all empty fields. 
```
|| ((empty($ex_radiologist)
                && empty($ex_pending)
                && empty($ex_curr_sites)))
```
But indeed, in case the requirements for data entry change in the form, probably best to simplify the 'or' part of the logic to `|| empty($ex_curr_sites)`

Thanks!

Reply:
This makes sense to me as well. Just to clean it up, this is how it should look like:
```
if (!empty($ex_radiologist)
    && !empty($ex_pending)
    && !empty($ex_curr_sites)
    && empty($examinerID)
    ) {
    	// The case in where examiner is being added for the first time
    	// 
    	...
      }
else if (!empty($examinerID)
		 && !empty($ex_pending) 
		 && !empty($ex_radiologist)) 
		 && (empty($ex_curr_sites) || !empty($ex_curr_sites))
		) {

		 	// The case where the examiner already exists 
		 	// and has either all sites being deleted or the sites are being changed
		  }
```

Reply:
Shouldn't the else if statement be:
```
else if (!empty($examinerID)
		 && !empty($ex_pending) 
		 && !empty($ex_radiologist)) 
		 && (empty($ex_curr_sites) || !empty($ex_curr_sites))
		) {

		 	// The case where the examiner already exists 
		 	// and has either all sites being deleted or the sites are being changed
		  }
```

Reply:
you are correct but that means that there is a bigger fautl in logic in the field validation in the front end. Thats def not within the scope of this PR since its not a bugfix but rather a minor revamp.

This should do for the moment. I will try to test as soon as I can.